lib/extensions/ar_virtual.rb

```diff
@@ -116,19 +116,30 @@ def virtual_delegate_arel(col, to, to_model, to_ref)
         if to_ref.macro == :has_one
           lambda do |t|
             src_model_id = arel_attribute(to_ref.association_primary_key, t)
-            to_model_id = to_model.arel_attribute(to_ref.foreign_key)
-            Arel.sql("(#{to_model.select(to_model.arel_attribute(col)).where(to_model_id.eq(src_model_id)).to_sql})")
+            VirtualDelegates.select_from_alias(to_model, to_ref, col, to_ref.foreign_key, src_model_id)
           end
         elsif to_ref.macro == :belongs_to
           lambda do |t|
-            src_model_id = arel_attribute(to_ref.association_foreign_key, t)
-            to_model_id = to_model.arel_attribute(to_ref.active_record_primary_key)
-            Arel.sql("(#{to_model.select(to_model.arel_attribute(col)).where(to_model_id.eq(src_model_id)).to_sql})")
+            src_model_id = arel_attribute(to_ref.foreign_key, t)
+            VirtualDelegates.select_from_alias(to_model, to_ref, col, to_ref.active_record_primary_key, src_model_id)
           end
         end
       end
     end
   end
+
+  def self.select_from_alias(to_model, to_ref, col, to_model_col_name, src_model_id)
```

Comment:
Minor but can you document this method?  Perhaps with examples that demonstrate inputs/outputs.  I think that would go a long way to helping me understand the use cases for this and why it fixes the problem in question.

I think what I'm concerned about is that all of those cases in the OP are handled in ActiveRecord already, so it feels like we are reinventing problems they have already solved.